requirements.txt

```diff
@@ -20,6 +20,7 @@ monai[nibabel]
 msvc-runtime==14.29.30133; sys.platform == 'win32' # append_to_freeze
 nibabel
 nilearn
+nnunetv2
```

Comment:
Since the major overhaul is reflected in the project name (`nnunetv1` versus `nnunetv2`), not just in the version number, this should be fine. Also, keep in mind that the version is unpinned on the `master` branch, but gets automatically fully pinned in `requirements-freeze.txt` whenever we create an SCT release.

Reply:
Note that for stable releases of SCT, the versions will still get pinned (via a separate `requirements-freeze.txt` file), so the question of "whether or not to pin dependencies" is mainly a question for people who install SCT via `master` (i.e. developers).

There are definitely pros and cons to pinning vs. not pinning on `master`, though!

<details>
<summary> Lots of thoughts! </summary>

- Leaving things pinned, as you suggest, minimizes the amount of urgent errors for developers when things change upstream. (We have had a lot of open issues that say something like "The tests are failing now due to an update!!")
- However, in many cases, leaving a package pinned is just postponing dealing with the problems until some later time. 
  - Upgrading Python versions often necessitates upgrading dependencies to be compatible with the new Python version. 
  - The various [maintenance PRs](https://github.com/ivadomed/ivadomed/pull/1308) we have had to work on in `ivadomed` lately also necessitated us upgrading packages, too.
- So, IMO, proactively pinning dependencies is a question of "do we want to deal with problems incrementally as they come up" vs. "do we want to deal with problems all at once, often when it's critically needed due to external pressure."
- Personally, I find "deal with issues as they come up" to be a little easier to work with? It makes me feel on top of things well in advance, as we're upgrading immediately (vs. being forced to upgrade later on). It also tends to isolate any issues to specific libraries, while leaving things pinned might result in interactions between multiple outdated libraries.
- There is also the benefit of automatic security upgrades / bugfixes that could theoretically mitigate future issues? But then again, there's also the risk of subtle behavior changes between versions, so maybe this point is a bit of a wash.

Also, some comments specific to `nnunet`:

- The rewrite seems to have been a one-off event, with [very understandable justification](https://github.com/MIC-DKFZ/nnUNet#what-happened-to-the-old-nnu-net). 
- Also, during the rewrite, they opted for a new package name altogether (`nnunet` vs. `nnunetv2`). Meaning, pinning the version wouldn't have had any effect in that specific case.

</details>

Ultimately, I think that either pinning or not pinning is OK. But, just to stick with what SCT has been doing so far, I think I'm leaning towards keeping it unpinned. :)

Reply:
Oops, GitHub didn't update to show @mguaypaq's response. But, I'm glad to see we both said many of the same things independently. :hearts:

Reply:
> but gets automatically fully pinned in requirements-freeze.txt whenever we create an SCT release.

cool! didn't know this, thanks for this info! I am fine with unpinning the version in any case!